python/ray/serve/_private/deployment_state.py

```diff
@@ -609,17 +637,23 @@ def check_stopped(self) -> bool:
             handle = ray.get_actor(self._actor_name, namespace=SERVE_NAMESPACE)
             stopped = check_obj_ref_ready_nowait(self._graceful_shutdown_ref)
             if stopped:
-                ray.kill(handle, no_restart=True)
                 try:
                     ray.get(self._graceful_shutdown_ref)
                 except Exception:
                     logger.exception(
                         "Exception when trying to gracefully shutdown replica:\n"
                         + traceback.format_exc()
                     )
+
+                ray.kill(handle, no_restart=True)
         except ValueError:
-            # ValueError thrown from ray.get_actor means actor has already been deleted
+            # ValueError thrown from ray.get_actor means actor has already been deleted.
             stopped = True
+        finally:
+            # Remove the placement group both if the actor has already been deleted or
```

Comment:
My understanding is that this removes the placement group as soon as the replica shuts down (e.g. if the replica is being killed as part of an in-place upgrade).

According to the note in [this section](https://docs.ray.io/en/releases-2.6.1/ray-core/scheduling/placement-group.html#remove-placement-groups-free-reserved-resources), other Ray actors/tasks that were co-scheduled with the replica using the PG also get stopped when the PG is removed (i.e. when the replica shuts down). Is that the expected behavior?

Reply:
Based on the unit tests, it looks like co-scheduling Ray actors/tasks with Serve replicas isn't the main purpose of this change. I'm not sure if my question is relevant. Feel free to resolve if it's not.